frontend/class-breadcrumbs.php

```diff
@@ -518,13 +518,26 @@ private function add_blog_crumb() {
 
 	/**
 	 * Add Blog crumb to the crumbs property for single posts where Home != blogpage.
+	 *
+	 * @return void
 	 */
 	private function maybe_add_blog_crumb() {
-		if ( ( 'page' === $this->show_on_front && 'post' === get_post_type() ) && ( ! is_home() && ! is_search() ) ) {
-			if ( $this->page_for_posts && WPSEO_Options::get( 'breadcrumbs-display-blog-page' ) === true ) {
-				$this->add_blog_crumb();
-			}
+		// When the show blog page is not enabled.
+		if ( WPSEO_Options::get( 'breadcrumbs-display-blog-page' ) !== true ) {
+			return;
 		}
+
+		// When there is no page configured as blog page.
+		if ( 'page' !== $this->show_on_front || ! $this->page_for_posts ) {
+			return;
+		}
+
+		// When the current page isn't a singular post.
+		if ( is_home() || is_search() || ( ! is_singular( 'post' ) ) ) {
```

Comment:
Not sure about this check:
- why the parenthesis around ` ( ! is_singular( 'post' ) )`  ?
- as far as I know, the search page is not a singular post
- same for the home?
- not 100% sure but maybe `is_home() || is_search()` could be removed?

Reply:
We can remove the parentheses

I have added the `is_home()` and 'is_search()' because it was used before.